state-chain/runtime/src/lib.rs

```diff
@@ -1427,9 +1427,15 @@ impl_runtime_apis! {
 
 		fn cf_scheduled_swaps(base_asset: Asset, _quote_asset: Asset) -> Vec<ScheduledSwap> {
 
+			let current_block = System::block_number();
+
 			pallet_cf_swapping::SwapQueue::<Runtime>::iter().flat_map(|(block, swaps_for_block)| {
+
+				// In case `block` has already passed, the swaps will be re-tried at the next block:
+				let execute_at = if block > current_block { block } else { current_block.saturating_add(1) };
```

Comment:
I think this would be a little easier to read as 
```suggestion
				let execute_at = core::cmp::max(block, current_block.saturating_add(1));
```